amazonfreertossdk/src/main/java/software/amazon/freertos/amazonfreertossdk/AmazonFreeRTOSDevice.java

```diff
@@ -937,15 +941,15 @@ private void publishToDevice(final Publish publish) {
     }
 
     private void discoverServices() {
-        if (mBleConnectionState == BleConnectionState.BLE_CONNECTED && mBluetoothGatt != null) {
+        if (isBLEConnected() && mBluetoothGatt != null) {
             sendBleCommand(new BleCommand(DISCOVER_SERVICES));
         } else {
             Log.w(TAG, "Bluetooth connection state is not connected.");
         }
     }
 
     private void setMtu(int mtu) {
-        if (mBleConnectionState == BleConnectionState.BLE_CONNECTED && mBluetoothGatt != null) {
+        if (mBleConnectionState == BleConnectionState.BLE_INITIALIZED && mBluetoothGatt != null) {
```

Comment:
`setMtu()` is called while BLE is being initialized so this will skip setting MTU. probably check `isBLEConnected()` as it is valid as soon as BLE is connected.